File: year2020/month11/day1123/ms_1123.cs
using System;

namespace algorithm_106
{
    class ms_1123
    {
        static bool isIN(String data, String find) 
        {
            if(data.IndexOf(find) != -1)
                return true;
            else
                return false;
        }

        static void Main(string[] args)
        {
            String[] data = new String[5];

            for(int i = 0; i < data.Length; i++)
            {
                String temp = Console.ReadLine();

                if(temp.Length <= 100)
                {
                    data[i] = temp;
                }
                else i--;
            }

            char c = Convert.ToChar(Console.ReadLine());
            String input = "";

            for(int i = 0; i < 1; i++)
            {
                String temp = Console.ReadLine();

                if(temp.Length <= 100)
                {
                    input = temp;
                }
                else i--;
            }

            bool isAny = false;

            for(int i = 0; i < data.Length; i++)
            {
                if(isIN(data[i], Convert.ToString(c)) || isIN(data[i], input)) 
                {
                    Console.WriteLine(data[i]);
                    isAny = true;
                }
            }

            if(!isAny)
                Console.WriteLine("none");
        }
    }
}

File: year2020/month11/day1123/ms_1123.cpp
#include <iostream>
#include <string>

using namespace std;

bool isIn(string data, string word);

int main() {
	string data[5];
	string input = "";
	string temp;
	char c;

	for (int i = 0; i < 5; i++) {
		getline(cin, temp);

		if (temp.length() <= 100) {
			data[i] = temp;
		}
		else i--;
	}

	cin >> c;
	getline(cin, temp); // fflush

	for (int i = 0; i < 1; i++) {
		getline(cin, temp);

		if (temp.length() <= 100) {
			input = temp;
		}
		else i--;
	}

	bool isAny = false;
	string str = "";

	str += c;

	for (int i = 0; i < 5; i++) {
		if (isIn(data[i], str) || isIn(data[i], input)) {
			cout << data[i] << endl;
			isAny = true;
		}
	}

	if (!isAny)
		cout << "none" << endl;

	return 0;
}

bool isIn(string data, string word) {
	int temp = data.find(word);

	if (0 <= temp && temp <= data.length()) {
		return true;
	}
	else return false;
}